src/muse/sectors/sector.py

```diff
@@ -103,25 +105,38 @@ def __init__(
         interpolation: str = "linear",
         outputs: Callable | None = None,
         supply_prod: PRODUCTION_SIGNATURE | None = None,
+        timeslice_level: str | None = None,
     ):
         from muse.interactions import factory as interaction_factory
         from muse.outputs.sector import factory as ofactory
         from muse.production import maximum_production
+        from muse.timeslices import TIMESLICE
 
-        self.name: str = name
         """Name of the sector."""
-        self.subsectors: Sequence[Subsector] = list(subsectors)
+        self.name: str = name
+
+        """Timeslice level for the sector (e.g. "month")."""
+        self.timeslice_level = timeslice_level or get_level(TIMESLICE)
+
         """Subsectors controlled by this object."""
-        self.technologies: xr.Dataset = technologies
+        self.subsectors: Sequence[Subsector] = list(subsectors)
+
         """Parameters describing the sector's technologies."""
+        self.technologies: xr.Dataset = technologies
+        if "timeslice" in self.technologies.dims:
+            if not get_level(self.technologies) == self.timeslice_level:
+                raise ValueError(
+                    f"Technodata for {self.name} sector does not match "
+                    "the specified timeslice level for that sector "
+                    f"({self.timeslice_level})"
+                )
```

Comment:
I agree with your implementation because it is explicit, but just in case, have you considered pulling the `timeslice_level` directly from the technodata rather than opening the possibility - that you check here - of having technodata timeslice information inconsistent with the declared timeslice level?

Reply:
This would be possible for sectors where timeslice-level utilization factor and/or minimum service factor are specified. However, for sectors where this isn't specified (i.e. using a single UF/MSF for all timeslices), there's no timeslice information in the technodata so nothing to pull from, and these sectors can take any timeslice level.

In the end, it's easier just to be explicit about it for all sectors.